examples/common/tracing/decoder/logging/ToCertificateString.cpp

```diff
@@ -0,0 +1,113 @@
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ *
+ */
+
+#include "ToCertificateString.h"
+
+#include <lib/support/Base64.h>
+#include <lib/support/ScopedBuffer.h>
+
+namespace {
+constexpr const uint8_t kChipRawPrefix[] = { 0x15, 0x30, 0x01 };
+
+bool IsChipCertificate(const chip::ByteSpan & data)
+{
+    return (data.size() > sizeof(kChipRawPrefix)) && (memcmp(data.data(), kChipRawPrefix, sizeof(kChipRawPrefix)) == 0);
+}
+
+const char * ToCertificate(const chip::ByteSpan & data, char * buffer, size_t len, const char * header = "",
+                           const char * footer = "")
+{
+    // Reset the buffer
+    memset(buffer, '\0', len);
+
+    if (data.size() == 0)
+    {
+        return buffer;
+    }
+
+    size_t base64DataLen = BASE64_ENCODED_LEN(data.size()) + 1;
+    if (base64DataLen + strlen(header) + strlen(footer) > len)
+    {
+        ChipLogError(DataManagement, "The certificate buffer is too small to hold the base64 encoded certificate");
+        return buffer;
+    }
+
+    chip::Platform::ScopedMemoryBuffer<char> str;
+    str.Alloc(base64DataLen);
+
+    auto encodedLen       = chip::Base64Encode(data.data(), data.size(), str.Get());
+    str.Get()[encodedLen] = '\0';
+
+    if (IsChipCertificate(data))
+    {
+        // Wherever Matter Operational Certificate Encoding representation is used, all certificates SHALL NOT be longer than 400
+        // bytes in their TLV form.
+        if (data.size() > 400)
+        {
+            ChipLogError(DataManagement, "Certificate size is greater than 400 bytes");
+        }
+
+        snprintf(buffer, len, "%s", str.Get());
+    }
+    else
+    {
+        // All certificates SHALL NOT be longer than 600 bytes in their uncompressed DER format.
+        if (data.size() > 600)
+        {
+            ChipLogError(DataManagement, "Certificate size is greater than 600 bytes");
+        }
+
+        size_t inIndex  = 0;
+        size_t outIndex = strlen(header) + 1;
+
+        snprintf(buffer, len, "%s\n", header);
+        for (; inIndex < base64DataLen; inIndex += 64)
+        {
+            outIndex += snprintf(&buffer[outIndex], len - outIndex, "%.64s\n", &str[inIndex]);
+        }
+        snprintf(&buffer[outIndex], len - outIndex, "%s", footer);
+    }
+
+    return buffer;
+}
+
+} // namespace
+
+namespace chip {
+namespace trace {
+namespace logging {
+
+const char * ToCertificateString(const ByteSpan & data, char * buffer, size_t len)
```

Comment:
I see buffer + len passed on here and in other places (ToCertificate).

Could we use `MutableCharSpan` instead?

Reply:
Done.